// jsdom/index.js
export class JSDOM {
  constructor(html = '<!DOCTYPE html><html><body></body></html>') {
    const document = createDocument();
    this.window = {
      document,
      Event: class {
        constructor(type) { this.type = type; }
      }
    };
  }
}

function createDocument() {
  const elementsById = {};

  function matches(el, selector) {
    if (selector.startsWith('.')) {
      const cls = selector.slice(1);
      return el.className.split(' ').includes(cls);
    }
    return el.tagName.toLowerCase() === selector.toLowerCase();
  }

  function querySelector(root, selector) {
    for (const child of root.children) {
      if (matches(child, selector)) return child;
      const found = querySelector(child, selector);
      if (found) return found;
    }
    return null;
  }

  function querySelectorAll(root, selector, acc = []) {
    for (const child of root.children) {
      if (matches(child, selector)) acc.push(child);
      querySelectorAll(child, selector, acc);
    }
    return acc;
  }

  function createElement(tag) {
    const el = {
      tagName: tag.toUpperCase(),
      children: [],
      className: '',
      textContent: '',
      title: '',
      disabled: false,
      id: '',
      parentElement: null,
      addEventListener(type, handler) {
        this._listeners = this._listeners || {};
        this._listeners[type] = handler;
      },
      dispatchEvent(evt) {
        if (this._listeners && this._listeners[evt.type]) {
          this._listeners[evt.type](evt);
        }
      },
      appendChild(child) {
        this.children.push(child);
        child.parentElement = this;
        if (child.id) elementsById[child.id] = child;
      },
      querySelector(selector) {
        return querySelector(this, selector);
      },
      querySelectorAll(selector) {
        return querySelectorAll(this, selector);
      },
      click() {
        this.dispatchEvent({ type: 'click' });
      }
    };
    return el;
  }

  const document = {
    createElement,
    getElementById(id) {
      return elementsById[id] || null;
    },
    body: createElement('body')
  };

  return document;
}
